### ISSUE TITLE
Persistent "connection failed" on only SOME autoupload files, retry does not hit the server

### ISSUE THREAD

**Issue by ohthehugemanatee** [nextcloud/android#10569]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Enable autoupload for photos
2. Take some photos
3. Observe "connection failed" for approx 1/8 of photos. 

### Expected behaviour

All photos should either upload or fail. Retry should make a visible HTTP request to the server.

### Actual behaviour

One photo will get stuck with "connection failed". Others taken before or after will upload successfully. Retrying the failed photo will always get the same "connection failed", but leaves no trace in nginx access.log or error.log. 

### Android version

12

### Device brand and model

samsung galaxy s21

### Stock or custom OS?

Stock

### Nextcloud android app version

30.20.3

### Nextcloud server version

24.0.2

### Using a reverse proxy?

Yes

### Additional information

If I manually "share" the photo to the camera uploads directory,  it works fine. Autoupload yhen correctly raises the conflict dialog. So it must be getting SOME information.

Sorry no logcat for now but i'll attach one the next time the issue appears.

**Comment by Nowaker** [comment]
I can confirm this happening to me as well. And I can confirm the workaround works too - removing the "failing" file from the autoupload list, then sharing it again to upload resolves the problem.

**Comment by AlvaroBrey** [comment]
Anything in common for the pictures that do fail, in contrast to the other ones? i.e. do they all have a name that starts in the same way, or something like that?

**Comment by Nowaker** [comment]
No, just regular pictures from the camera, therefore following the standard auto-generated name based on date. 99.5% of such pictures just go through, while a couple bad sheep simply fail and won't upload unless I "Clear failed uploads" and then share them manually from my gallery into Nextcloud.

**Comment by poikilos** [comment]
If I share one file from outside the app (such as from the Photos app), it always succeeds. If I share multiple files from the photos app, only the first one succeeds and each of the others have the error. I can't seem to find any permission for allowing it and I'm not sure whether it is a permission issue. If I go to the Nextcloud app and tap each of the failed ones, they always succeed (unless I try it days later...). This has happened to me for years across several updates of Android and the Nextcloud app. Is my issue the same or different?

**Comment by juliuspopp** [comment]
I have this issue with my note files (.md files from Obsidian). Most are auto uploaded, some constantly fail and cause the app to crash

**Comment by joshtrichards** [comment]
Are any of you still experiencing this?

**Comment by Poikilos** [comment]
- How to help make sure it is fixed: Connect to Wi-Fi or turn on mobile data. Open Nextcloud app and make sure it sees your data, then exit that app. Go to your Android photo gallery, long press a photo then tap at least 2 others to select multiple photos at once, share, Nextcloud, select a folder, confirm. Wait 10 seconds. Go to Nextcloud app, click ☰ menu button at top left, Uploads, and usually one says uploaded and the rest failed. If any failed, please report it here and state your Nextcloud app version and Android version.
- App version 3.22.3 on Android 14 does not have the problem for me.

I am bumping this only because the issue was very bad and in one case I deleted files I thought I shared but had failed. Yeah, I should have validated the "backup" before depending on it, but whether I lost data or not the same amount of time is wasted going to my computer and waiting for the files to appear, then going to the web interface on said computer, and seeing they aren't there either.

Bottom line: If you test again it may work (good), and if not then you will help the issue get fixed if you report if it is still happening on your version.

**Comment by Nowaker** [comment]
> Are any of you still experiencing this?

@joshtrichards Yes, continuously encountering it.
